drivers/flash/flash_shell.c

```diff
@@ -52,22 +43,28 @@ static int cmd_erase(const struct shell *shell, size_t argc, char *argv[])
 	if (argc > 2) {
 		size = strtoul(argv[2], NULL, 16);
 	} else {
-#if defined(CONFIG_SOC_FLASH_NRF)
-		size = NRF_FICR->CODEPAGESIZE;
-#else
-		error(shell, "Missing size.");
-		return -EINVAL;
-#endif
+		struct flash_pages_info info;
+
+		result = flash_get_page_info_by_offs(flash_dev, page_addr,
+						     &info);
+
+		size = info.size;
```

Comment:
While it doesn't change the compiled code, you might want to assign sign after checking for error.

Reply:
oops there is a typo. I meant assign size not sign.